cataclysmbnteam/Cataclysm-BN
# feat(balance): Redesign zombie masters

## Checklist



### Required

- [x] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [x] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

### Optional



## Purpose of change

Making zombie masters no longer "player-dependent" to function correctly. They're very strong if left in the bubble for long, but meaningless otherwise.



## Describe the solution

Zombie masters no longer upgrade zombies. Instead they redirect nearby (range 30, 40 if attacked) zombies to their own target and give them a moderate buff (20 speed, 1 to-hit, 2 dodge).
This has a long (100) cooldown, but the cooldown is only applied on attack version, not on defense.



## Describe alternatives you've considered

* Removing zombie masters
* Giving zombie masters summon ability instead of buff ability



## Testing



* Spawn zombie master and lots of zombies around it
* Set time to night, so that zombies can't see player, but master can (range 15)
* Wait for it to point at the player
* Zombies should become "empowered" and go towards player

## Additional context
